examples/plot_pp_study.py

```diff
@@ -0,0 +1,362 @@
+#       - Median value of owner-occupied homes in $1000's
+# *Figure A: Dataset variables*
+#
+# Where MEDV (colored green in Figure A above) is the target variable
+# (that a model built on this dataset is intended to predict). We make
+# modifications below with the understanding that LSTAT is a parameter of
+# concern.
+#
+# **Import and install relevant packages**
+#
+import numpy as np
+import pandas as pd
+from fairlearn.metrics import MetricFrame
+from fairlearn.metrics import selection_rate
+from fairlearn.reductions import DemographicParity, ExponentiatedGradient
+from fairlearn.datasets import fetch_boston
+from sklearn.metrics import accuracy_score
+from sklearn.tree import DecisionTreeClassifier
+from statistics import mean, median
+from fairlearn.widget import FairlearnDashboard
+
+# %%
+#
+# **1. We devise a new target column based on CRIM (colored blue in Figure A above), indicating whether a certain per capita crime rate threshold is met.**
+#
+#    While we calculate and print Mean, Median, Max, and Min crime
+#    rates below, the analysis to follow is specifically for the mean crime
+#    rate as the chosen threshold. Similar results are observed for median
+#    crime rate as the chosen threshold as well (which one can observe by
+#    changing the index of threshold_crime_rates to 1).
+#
+bostonDS = fetch_boston(as_frame=True)
+crimerate_data = bostonDS.data['CRIM']
+
+mean_crimerate = mean(crimerate_data)
+median_crimerate = median(crimerate_data)
+threshold_crime_rates = [mean_crimerate, median_crimerate]
+min_crimerate = min(crimerate_data)
+max_crimerate = max(crimerate_data)
+"""
+[0] = Mean
+[1] = Median
+"""
+y_true = (crimerate_data > threshold_crime_rates[0]) * 1
+
+rawData = bostonDS.data
+print(rawData)
+print('Min crime rate', min_crimerate)
+print('Max crime rate', max_crimerate)
+print('Mean crime rate', mean_crimerate)
+print('Median crime rate', median_crimerate)
+# %%
+#
+# **2. We build a classifier using CRIM as the target value that the model ultimately predicts, and also removing CRIM in the data used to train the model.**
+#
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+
+data_without_crimerate = rawData.drop('CRIM', 1)
+
+print(data_without_crimerate)
+
+# %%
+#
+
+classifier.fit(data_without_crimerate, y_true)
+
+y_pred = classifier.predict(data_without_crimerate)
+# %%
+#
+# **3. We create bins for LSTAT (colored blue in Figure A above), to understand the models performance on different LSTAT value intervals.**
+#
+#  While we calculate and print Mean, Median, Max, and Min LSTAT
+#  values, the chosen bins are based on the observe Min and Max LSTAT
+#  values.
+#
+lstat = bostonDS.data['LSTAT']
+mean_lstat = mean(lstat)
+median_lstat = median(lstat)
+max_lstat = max(lstat)
+min_lstat = min(lstat)
+
+print('Mean LSTAT ', mean_lstat)
+print('Median LSTAT ', median_lstat)
+print('Max LSTAT ', max_lstat)
+print('Min LSTAT ', min_lstat)
+
+bins = [0, 10, 20, 30, 40]
```

Comment:
Can these be computed based on the min and max LSTAT? Otherwise these look a bit like magic numbers.